common/src/main/java/co/elastic/otel/common/LocalRootSpan.java

```diff
@@ -93,31 +106,14 @@ public static void onSpanStart(ReadableSpan startedSpan, Context parentContext)
    * </ul>
    */
   @Nullable
-  public static Span getFor(Span span) {
-    return resolveRoot(span, localRoot.get(span));
+  public static ReadableSpan getFor(Span span) {
+    return getFor((ReadableSpan) span);
   }
 
   /** See {@link LocalRootSpan#getFor(Span)}. */
   @Nullable
-  public static Span getFor(ReadableSpan span) {
-    return resolveRoot(span, localRoot.get(span));
-  }
-
-  /** See {@link LocalRootSpan#getFor(Span)}. */
-  @Nullable
-  public static Span getFor(ReadWriteSpan span) {
-    return resolveRoot(span, localRoot.get(span));
-  }
-
-  @Nullable
-  private static Span resolveRoot(Object span, Object rootSpanVal) {
-    if (rootSpanVal == LOCAL_ROOT_MARKER) {
-      return (Span) span;
```

Comment:
This cast would cause an exception if the provided span is a `MutableSpan`. That's why I changed `LocalRootSpan` to operate on `ReadableSpan` instead of `Span`.